Week 2/Weekly Practice Problems (Week 2)/2553: Separate the Digits in an Array/delight2553.java

```diff
@@ -0,0 +1,43 @@
+/*Solution 1 using String*/
+class Solution {
+    public int[] separateDigits(int[] nums) {
+        String temp;
+        ArrayList<Integer> list = new ArrayList<>();
+        for(int i = 0; i < nums.length; i++){
+            temp = Integer.toString(nums[i]);
+            for(int j = 0; j < temp.length(); j++){
+                list.add(temp.charAt(j)-'0');
+            }    
+        }
+        int[] returnArray = new int[list.size()];       
```

Comment:
I believe there's an toArray method on array list!

Reply:
I wanted to use toArray function(list.toArray()), but that returned Object[] and I didn't quite get how to convert that into int[], so I just used the for loop to manually copy the list into an int array. How would you use the toArray() in this case?

Reply:
You can do something like this:

return list.toArray(new Int[list.size()]);